### PR TITLE
Pretty Print Value of Public Fields in FSI

### LINKED ISSUE #1382: FSI Mutable Struct Field Pretty Printing
Structs that contain mutable fields do not render in FSI, the data is still there, and everything is fine if we try to access the field via FSI, it is only the presentation in FSI that seems to be inconsistent. Unsure if bug or limitation. 

#### Repro steps
```
// Firstly, define the mutable struct.
[<Struct>]
type t = val mutable x : int
// create a reference to it
let mutable t = t()
// set the field
t.x<-1
// now try to access the pretty representation of the struct
t;;
// prints val it : t = FSI_0094+t
```

Further, if we define a normal ctor'd struct with initialised values and a mutable field, we get a ghost field:
```
[<Struct>]
type test2 = 
    val x : int
    val mutable y : int
    new (x) = {x=x; y = 1 }

let t2 = test2(1)
t2;;
// val it : test2 = FSI_0097+test2 {x = 1;}
```

#### Expected behavior

Expected behaviour is that a mutable struct prettyprints itself in the same manner as a non mutable struct.

#### Related information 

Provide any related information 

* Latest F# via Visual Studio 2015 update 3

Edit:made struct attributes consistent

### PR DESCRIPTION
**Pull request by liboz**
Fixes #1382. Also has some minor cleanup

### CODE REVIEW

**Comment by liboz** [comment]
Ready for review

File: src/utils/sformat.fs [hunk 1]
```diff
@@ -862,7 +864,7 @@ namespace Microsoft.FSharp.Text.StructuredFormat
                                         let postText = m.Groups.["post"].Value // Everything after the closing bracket
```
Comment (KevinRansom) [inline src/utils/sformat.fs]: Up at line __832__:
we get x.GetType() again ... I suppose we can remove it and use the ty that we fetched in __827__
````
    let ty = x.GetType()
        if ty.IsGenericType && ty.GetGenericTypeDefinition() = typedefof<Lazy<_>> then
            Some (wordL (x.ToString()))
        else
            // Try the StructuredFormatDisplayAttribute extensibility attribute
            match x.GetType().GetCustomAttributes (typeof<StructuredFormatDisplayAttribute>, true) with
````

File: src/utils/sformat.fs [hunk 2]
```diff
@@ -1114,17 +1119,21 @@ namespace Microsoft.FSharp.Text.StructuredFormat
                               // massively reign in deep printing of properties 
                               let nDepth = depthLim/10
 #if FX_ATLEAST_PORTABLE
-                              System.Array.Sort((props),{ new System.Collections.Generic.IComparer<PropertyInfo> with member this.Compare(p1,p2) = compare (p1.Name) (p2.Name) } );
+                              System.Array.Sort((propsAndFields),{ new System.Collections.Generic.IComparer<MemberInfo> with member this.Compare(p1,p2) = compare (p1.Name) (p2.Name) } );
```
Comment (KevinRansom) [inline src/utils/sformat.fs]: we do an open System ... so I guess we can do Array.Sort here instead of namespace qualifying..

**Comment by KevinRansom** [comment]
@liboz 
Other than the minor comments it looks great.

Kevin

**Comment by KevinRansom** [comment]
@liboz 

This is great Thank you for taking care of this.

Kevin

**Pull request merged**